### PR TITLE
Tests for persistence exclusion configuration

### PR DESCRIPTION
**Pull request by mherwege**
Related to https://github.com/openhab/openhab-core/pull/4468

@holgerfriedrich for your review

### CODE REVIEW

File: bundles/org.openhab.core.persistence/src/test/java/org/openhab/core/persistence/internal/PersistenceManagerTest.java [hunk 1]
```diff
@@ -235,10 +237,52 @@ public void appliesToItemWithAllConfig() {
...
+        addConfiguration(TEST_PERSISTENCE_SERVICE_ID,
+                List.of(new PersistenceAllConfig(), new PersistenceGroupExcludeConfig(TEST_GROUP_ITEM_NAME)),
+                PersistenceStrategy.Globals.UPDATE, null);
+
+        manager.stateUpdated(TEST_ITEM, TEST_STATE);
+        manager.stateUpdated(TEST_ITEM2, TEST_STATE);
+        manager.stateUpdated(TEST_GROUP_ITEM, TEST_STATE);
+
+        verify(persistenceServiceMock).store(TEST_ITEM2, null);
+        verify(persistenceServiceMock).store(TEST_GROUP_ITEM, null);
```
Comment (holgerfriedrich) [inline bundles/org.openhab.core.persistence/src/test/java/org/openhab/core/persistence/internal/PersistenceManagerTest.java:277]: Not sure if I get this test correctly. Could you pls. explain why the group item is here, and Item1 is excluded?

Comment (mherwege) [reply bundles/org.openhab.core.persistence/src/test/java/org/openhab/core/persistence/internal/PersistenceManagerTest.java:277]: States of TEST_ITEM, TEST_ITEM2 and TEST_GROUP_ITEM are updated. The persistence configuration is to persist all except items inside TEST_GROUP_ITEM. TEST_ITEM is in TEST_GROUP_ITEM.
Therefore, the right behaviour is to persist TEST_ITEM2 and TEST_GROUP_ITEM itself (not its members).

Comment (holgerfriedrich) [reply bundles/org.openhab.core.persistence/src/test/java/org/openhab/core/persistence/internal/PersistenceManagerTest.java:277]: > TEST_ITEM is in TEST_GROUP_ITEM. 

This is exactly what I was looking for in the code. Got it now. Don't know why I overlooked line 143.

Thanks.

**Review by holgerfriedrich — APPROVED** [review]
LGTM, thanks!

**Pull request merged**